### DISCUSSION TITLE
How to check faulty sectors for other miners in the network

### DISCUSSION THREAD

**Discussion by LexLuthr** [filecoin-project/lotus#7728] [Tutorials]
The lotus-shed allows the miner in the network to check faulty sectors by other miners. The utility only requires the ID of the miner you wish to check for faulty sectors.

```
# lotus-shed miner faults -h
NAME:
   lotus-shed miner faults - Display a list of faulty sectors for a SP

USAGE:
   lotus-shed miner faults [command options] [minerAddress]

OPTIONS:
   --expiring-in value  only list sectors that are expiring in the next <n> epochs (default: 0)
   --help, -h           show help (default: false)
```
Example output:

```
# lotus-shed miner faults f049911
faulty sectors: 102707 102761 103178 104252 212630 213669 213671 214432 214442 216264 331374 331421 331819 331947 332323 440246 441364 441958 441973 442136 442146 442978 443341 546508 559341 560135 560219 560792 560848 560936 561040 561139 561150 561243 561346 561504 561947 563285 686703 689081 690499 690890 691191 691269 691303 692555 692793 692802 692860 693462 693571 693577 693836 693941 694130 694225 694295 694438 805682 806380 806774 807057 807602 874276 874824 875251 993678 994822 1108108 1109599 1110035 1110816 1225373 1226223 1226414 1227592 1335632 1336418 1337859 1340677 1341444 2028516 2029995

# lotus-shed miner faults f0142720
no fault

```
The command reports all the faulty sectors found on the miner or no fault in case all sectors are good.

**Comment by hcm609** [comment]
how can I let the miner to recheck and rebuild them so as to be ready to be submitted on chain?

**Reply by LexLuthr** [reply]
@hcm609 Right now, there is no process to send that info on the chain. It has to be done off the chain.

**Reply by hcm609** [reply]
off chain yes. But how can I force this? Does lotus-shed have such capability?

**Reply by LexLuthr** [reply]
@hcm609 No, lotus-shed does not have any such capabilities yet. The premise is intriguing but use case would be very limited as it would require miners to actively monitor other miners and inform them of faulty sectors(usually monitored by miners themselves). So, even if we inform them, there are chances that it will have no outcome if faults are due to a problem in miner. Another thing to note is that this information is being fetched from the chain so fault has already been reported.

**Reply by hcm609** [reply]
I get what you say but here is the thing. My own miner crashed or malfunctioned with the new 1.13.2-rc1 and I had to restart it and took 20minutes. Then I got the following:
Committed | 36.38 TiB
Proving | 35 TiB (32.81 TiB Faulty, 90.21%)
Those sectors do exist and I have no way to push or any script to write to so a rechecking of the 32TiB of sectors.
Any ideas?

**Reply by LexLuthr** [reply]
@hcm609 Would you be okay to take this conversation to slack? It would be easier and faster to find you some help there.

Under Filecoin slack channel #fil-lotus-help

**Reply by hcm609** [reply]
I can not join slack without an invitation. do you mean discord?

**Reply by LexLuthr** [reply]
https://docs.filecoin.io/community/chat-and-discussion-forums/
